src/south/ADS/ADS.class.js

```diff
@@ -0,0 +1,99 @@
+const ads = require('ads-client')
+
+const ProtocolHandler = require('../ProtocolHandler.class')
+
+/**
+ * Class ADS - Provides instruction for Modbus client connection
+ */
+class ADS extends ProtocolHandler {
+  /**
+   * Constructor for ADS
+   * @constructor
+   * @param {Object} dataSource - The data source
+   * @param {Engine} engine - The engine
+   * @return {void}
+   */
+  constructor(dataSource, engine) {
+    super(dataSource, engine)
+    this.connected = false
+    this.netId = this.dataSource.ADS.netId
+    this.port = this.dataSource.ADS.port
+  }
+
+  /**
+   * Runs right instructions based on a given scanMode
+   * @param {String} scanMode - Cron time
+   * @return {void}
+   */
+  /**
+   * On scan.
+   * @param {String} scanMode - The scan mode
+   * @return {Promise<void>} - The on scan promise
+   */
+  async onScan(scanMode) {
+    if (!this.connected) {
+      this.logger.silly(`onScan ignored: connected: ${this.connected}`)
+      return
+    }
+
+    const nodesToRead = this.dataSource.points.filter((point) => point.scanMode === scanMode)
+    if (!nodesToRead.length) {
+      this.logger.error(`onScan ignored: no points to read for scanMode: ${scanMode}`)
+      return
+    }
+
+    await nodesToRead.forEach((node) => {
+      this.client.readSymbol(node.address)
+        .then((nodeResult) => {
+          this.addValues([
+            {
+              pointId: node.pointId,
+              timestamp: new Date().toISOString(),
+              data: { value: JSON.stringify(nodeResult) },
```

Comment:
timestamp is never given by ADS source???

Reply:
From what I understand, we can get the date with subscription (to be implemented with listen mode later maybe).

Otherwise, we get the data at the date we request it, so this timestamp is correct. The timestamp is not returned by the payload.